src/sentry/api/serializers/models/incidentactivity.py

```diff
@@ -6,26 +6,36 @@
     Serializer,
     register,
 )
-from sentry.incidents.models import (
-    Incident,
-    IncidentActivity,
-)
+from sentry.utils.snuba import SnubaTSResult
+from sentry.api.serializers.snuba import SnubaTSResultSerializer
+from sentry.incidents.models import IncidentActivity
+from sentry.utils.db import attach_foreignkey
 
 
 @register(IncidentActivity)
 class IncidentActivitySerializer(Serializer):
     def get_attrs(self, item_list, **kwargs):
-        incidents = Incident.objects.filter(id__in=set(i.incident_id for i in item_list))
-        incident_lookup = {incident.id: incident for incident in incidents}
+        attach_foreignkey(item_list, IncidentActivity.incident, related=('organization',))
+        attach_foreignkey(item_list, IncidentActivity.event_stats_snapshot)
+        return {}
 
-        results = {}
-        for activity in item_list:
-            results[activity] = {'incident': incident_lookup[activity.incident_id]}
+    def serialize(self, obj, attrs, user):
+        incident = obj.incident
 
-        return results
+        event_stats = None
+        if obj.event_stats_snapshot:
+            serializer = SnubaTSResultSerializer(
+                obj.incident.organization,
+                None,
+                user,
+            )
+            event_stats = serializer.serialize(SnubaTSResult(
```

Comment:
Does this mean we get a snapshot for every activity type?

Reply:
Mostly not, because most activities won't have one. The key will always be available, it will just be null most of the time.